android/app/src/main/java/poke/rogue/helper/presentation/dex/model/SkillListItem.kt

```diff
@@ -0,0 +1,54 @@
+package poke.rogue.helper.presentation.dex.model
+
+import poke.rogue.helper.data.model.PokemonSkill
+import poke.rogue.helper.data.model.SkillCategory
+import poke.rogue.helper.presentation.type.model.TypeUiModel
+import poke.rogue.helper.presentation.type.model.toUi
+
+sealed class SkillListItem {
+    abstract val viewType: Int
+
+    data class SectionTitle(val title: String) : SkillListItem() {
+        override val viewType: Int = VIEW_TYPE_SECTION_TITLE
+    }
+
+    data object Header : SkillListItem() {
+        override val viewType: Int = VIEW_TYPE_HEADER
+    }
+
+    data class Skill(
+        val id: String,
+        val name: String,
+        val level: Int,
+        val power: String,
+        val type: TypeUiModel,
+        val accuracy: String,
+        val category: SkillCategory,
+    ) : SkillListItem() {
+        override val viewType: Int = VIEW_TYPE_SKILL
+
+        companion object {
+            const val NO_POWER = "-"
+            const val NO_ACCURACY = "-"
+        }
+    }
+
+    companion object {
+        const val VIEW_TYPE_SECTION_TITLE = 0
+        const val VIEW_TYPE_HEADER = 1
+        const val VIEW_TYPE_SKILL = 2
+    }
+}
```

Comment:
abstract val 를 사용하기 보다는 생성자로 넣어주는건 어떠신가요?

```suggestion
sealed class SkillListItem(val viewType: Int) {

    data class SectionTitle(val title: String) : SkillListItem(VIEW_TYPE_SECTION_TITLE)

    data object Header : SkillListItem(VIEW_TYPE_HEADER)

    data class Skill(
        val id: String,
        val name: String,
        val level: Int,
        val power: String,
        val type: TypeUiModel,
        val accuracy: String,
        val category: SkillCategory,
    ) : SkillListItem(VIEW_TYPE_SKILL) {
        companion object {
            const val NO_POWER = "-"
            const val NO_ACCURACY = "-"
        }
    }

    companion object {
        const val VIEW_TYPE_SECTION_TITLE = 0
        const val VIEW_TYPE_HEADER = 1
        const val VIEW_TYPE_SKILL = 2
    }
}
```

Reply:
오 그게 좋겠다. 좋은데요?